src/Logic.hs

```diff
@@ -311,11 +315,20 @@ handlePullRequestCommitChanged prId newSha state =
       -- If the pull request was not present in the first place, do nothing.
       Nothing -> pure state
 
+-- | Describe what caused the PR to close.
+prClosingMessage :: PRCloseCause -> Text
+prClosingMessage User = "Abandoning this pull request because it was closed."
+prClosingMessage StopIntegration = "Stopping integration because the PR changed after approval."
+
+-- | Handle PR close when a user actually closes a PR.
 handlePullRequestClosed :: PullRequestId -> ProjectState -> Action ProjectState
-handlePullRequestClosed pr state = Pr.deletePullRequest pr <$>
+handlePullRequestClosed = handlePullRequestClosed' User
```

Comment:
It would be better to call this `handlePullRequestClosedByUser`, because a single apostrophe does not tell the reader what the difference between the functions is.